addons/project_mail_plugin/controllers/project_client.py

```diff
@@ -0,0 +1,41 @@
+
+from odoo import http
+from odoo.http import request
+from odoo.tools import html2plaintext
+
+
+class ProjectClient(http.Controller):
+    @http.route('/mail_plugin/project/search', type='json', auth='outlook', cors="*")
+    def projects_search(self, search_term, limit=5):
+
+        projects = request.env['project.project'].search([('name', 'ilike', search_term)], limit=limit)
+
+        return [
+            {
+                'name': project.name,
+                'project_id': project.id,
+            }
+            for project in projects
+        ]
+
+    @http.route('/mail_plugin/task/create', type='json', auth='outlook', cors="*")
+    def task_create(self, email_subject, email_body, project_id, partner_id):
+
+        partner = request.env['res.partner'].browse(partner_id).exists()
+        if not partner:
+            return {'error': 'partner_not_found'}
+
+        record = request.env['project.task'].create({
+            'name': html2plaintext(email_subject),
+            'partner_id': partner_id,
+            'description': html2plaintext(email_body),
+            'project_id': project_id,
```

Comment:
I wonder if we should check that the user has access to this project too?
Maybe by browsing the project, the same way as partner, and returning an error of some kind ("project_not_found" ?).

Reply:
As the user can only select projects that he is related to (via the search dropdown), I don't think that this can be an issue for the create task route.

However, for the fetch tasks route, this could be problematic since the user can see a Task belonging to a Project for which he has no access if the task has been assigned to him, with the current implementation, this leads to a server error.

To overcome this issue, since on the Task form the user can see to which project it's related, I think that there is no harm into displaying the project name with the Task and for that we can use a sudo, can you tell me what do you think about?

I have tagged you in the code portion responsible for this.

Reply:
Even if the user can only select projects he has access to in the dropdown, I assume it would not be too hard if someone wanted to alter the project_id sent by your module and use this route to create some kind of attack?

I would say better safe than sorry in any case.

Reply:
Yeah that's right in this case.

I have added a check for this.